extmod/uasyncio/funcs.py

```diff
@@ -62,6 +62,9 @@ def remove(t):
 
 
 async def gather(*aws, return_exceptions=False):
+    if not aws:
+        return aws
```

Comment:
CPython returns an empty list in such a case.  This will return an empty tuple.

But I think that's fine.  It's an edge case, and if it is a problem (eg user tries to append to the return value) it will be a very visible error.

Reply:
The idea was to avoid an alloc. Seeing as there's already an .exp file due to other CPython diffs I didn't think it was important, although I considered changing it to `print(len(gather())` instead.

But you're right, I hadn't considered that the user might append to it... I guess I can imagine a scenario where that might be useful...

Reply:
Just to give a bit of context here the `aws` is dynamic based on several rules so it might or might not have "gatherables". And I think the previous behavior was returning an empty list anyway.